packages/core/handlers/integration-event-dispatcher.js

```diff
@@ -0,0 +1,55 @@
+
+/**
+ * Lightweight dispatcher that executes integration event handlers.
+ * Signature: `new IntegrationEventDispatcher(integrationInstance)`
+ * @param {import('../integrations/integration-base')} integrationInstance Pre-instantiated integration.
+ */
+class IntegrationEventDispatcher {
+  constructor(integrationInstance) {
+    if (!integrationInstance) {
+      throw new Error('Integration instance is required');
+    }
+    this.integrationInstance = integrationInstance;
+  }
+
+  async dispatchHttp({ event, req, res, next }) {
+    const instance = this.integrationInstance;
+
+    const handler = this.findEventHandler(instance, event);
+
+    if (!handler) {
+      const name = instance.constructor?.Definition?.name || 'integration';
+      throw new Error(`Event ${event} not registered for ${name}`);
+    }
+
+    return await handler.call(instance, { req, res, next });
+  }
+
+  async dispatchJob({ event, data, context }) {
+    const instance = this.integrationInstance;
+
+    const handler = this.findEventHandler(instance, event);
+
+    if (!handler) {
+      const name = instance.constructor?.Definition?.name || 'integration';
+      throw new Error(`Event ${event} not registered for ${name}`);
+    }
+
+    return await handler.call(instance, { data, context });
```

Comment:
Probably a linting thing at a later date but return await in an async function is unnecessary, so says my linters that bark at me